### PR TITLE
[shopsys] improved upgrade notes before new release

### PR DESCRIPTION
**Pull request by grossmannmartin**
| Q             | A
| ------------- | ---
|Description, reason for the PR| Improved upgrade notes based on the experience from the upgrade of the internal project
|New feature| No 
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| No 
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

File: upgrade/UPGRADE-v10.0.0-dev.md [hunk 1]
```diff
@@ -126,23 +159,13 @@ There you can find links to upgrade notes for other versions too.
...
-        - if you use default Shopsys Framework configuration, you can just use `ecs.php` (see #project-base-diff)
-        - for your custom configuration, you can leverage https://github.com/symplify/config-transformer
-    - see #project-base-diff to update your project 
-- **\[BC break\]** dropped support for unsupported PHP in packages ([#2416](https://github.com/shopsys/shopsys/pull/2416))
-    - packages `shopsys/migrations`, `shopsys/coding-standards`, and `shopsys/http-smoke-testing` no longer support PHP 7.2 and 7.3
-    - support for PHP 8.0 and 8.1 added to `shopsys/coding-standards` and `shopsys/http-smoke-testing`
-- **\[BC break\]** upgrade `"doctrine/migrations` and `doctrine/doctrine-migrations-bundle` ([#2414](https://github.com/shopsys/shopsys/pull/2414))
-    - see #project-base-diff to update your project
+    - if you have your own cache implemented with `Doctrine\Common\Cache` you may look at the [upgrade cache example](./upgrade-instructions-for-caches.md)
+- upgrade `"doctrine/migrations` and `doctrine/doctrine-migrations-bundle` ([#2414](https://github.com/shopsys/shopsys/pull/2414))
```
Comment (vitek-rostislav) [inline upgrade/UPGRADE-v10.0.0-dev.md:163]:
```suggestion
- upgrade `doctrine/migrations` and `doctrine/doctrine-migrations-bundle` ([#2414](https://github.com/shopsys/shopsys/pull/2414))
```

File: upgrade/upgrade-instructions-for-caches.md [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+  {
+-     $this->cacheProvider = $cacheProvider;
++     $this->cache = $cache;
+      // ... set properties
+  }
+```
+
+And don't forget about the `getBestsellingProductIds` method:
+
+```php
```
Comment (vitek-rostislav) [inline upgrade/upgrade-instructions-for-caches.md:150]: do not you want to show this as diff as well? Or did you only want to show the new implementation? The truth is, the diff would be probably hard to read, right? So maybe I would extend the sentence above with "The new implementation looks like this:"

File: upgrade/upgrade-instructions-for-caches.md [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+      // ... dependencies
+  )
+  {
+-     $this->cacheProvider = $cacheProvider;
++     $this->cache = $cache;
+      // ... set properties
+  }
+```
+
+And don't forget about the `getBestsellingProductIds` method:
```
Comment (vitek-rostislav) [inline upgrade/upgrade-instructions-for-caches.md:148]:
```suggestion
And don't forget about the `getBestsellingProductIds` method. The new implementation looks like this:
```

**Review by vitek-rostislav — CHANGES_REQUESTED** [review]

File: upgrade/UPGRADE-v10.0.0-dev.md [hunk 4]
```diff
@@ -1,19 +1,89 @@
...
+- class `\Shopsys\FrameworkBundle\Model\Customer\User\CustomerUser` no longer implements `Serializable` interface ([#2431](https://github.com/shopsys/shopsys/pull/2431))
+    - public methods `serialize()` and `unserialize()` were removed
+    - `__serialize()` and `__serialize()` are used instead
+- class `\Shopsys\FrameworkBundle\Model\Administrator\Administrator` no longer implements `Serializable` interface ([#2431](https://github.com/shopsys/shopsys/pull/2431))
+    - public methods `serialize()` and `unserialize()` were removed
+    - `__serialize()` and `__unserialize()` are used instead
+- allow running npm scripts even when they are not executable ([#2403](https://github.com/shopsys/shopsys/pull/2403))
+    - see [project-base-diff](https://github.com/shopsys/project-base/commit/76cfa5d35b11689ffdd126e3ca2079acbefb659f) to update your project
+
+## Upgrade composer dependencies and PHP version
```
[the remark this answers is not in the document]
Comment (grossmannmartin) [reply upgrade/UPGRADE-v10.0.0-dev.md:15]: it should. I have moved the `Application` section at the end

File: upgrade/UPGRADE-v10.0.0-dev.md [hunk 5]
```diff
@@ -1,19 +1,89 @@
...
+
+The main highlight of this version is upgrade of libraries and switch to PHP 8.1.
+The first you should do is update your composer.json to have installable set on PHP 8.1 compatible with the new version of Shopsys Framework.
+Pay special attention to any of your custom libraries as it's possible you will have to upgrade them too.  
+All the changes are considered as backwards incompatible.
+
+<!--- TODO change master to released version in the link  --->
+- upgrade PHP version to 8.1 ([#2430](https://github.com/shopsys/shopsys/pull/2430))
+    - update `docker/php-fpm/Dockerfile` and `scripts/install-docker-wsl-debian.sh` according to [project-base-diff](https://github.com/shopsys/project-base/commit/8ddaab98a3d3392c13735541505202cb63dfa5ce)
+    - apply all [composer.json changes](https://github.com/shopsys/project-base/compare/v9.1.2...master#diff-d2ab9925cad7eac58e0ff4cc0d251a937ecf49e4b6bf57f8b95aab76648a9d34) from this version at once
```
[the remark this answers is not in the document]
Comment (grossmannmartin) [reply upgrade/UPGRADE-v10.0.0-dev.md]: I have created a gist with the changes and added a todo to regenerate this gist during release

File: upgrade/upgrade-instructions-for-caches.md (hunk 2 above)
Comment (grossmannmartin) [reply upgrade/upgrade-instructions-for-caches.md:150]: Yup, the diff was truly hard to read. A have added the sentence 👍

File: upgrade/upgrade-instructions-for-caches.md (hunk 3 above)
Comment (grossmannmartin) [reply upgrade/upgrade-instructions-for-caches.md]: oh... I made the changes before I notice the suggestion...

Comment (vitek-rostislav) [reply upgrade/upgrade-instructions-for-caches.md]: yeah, I wrote the comment first and then added the suggestion :smile:

**Review by vitek-rostislav — APPROVED** [review]

**Pull request merged**